eth/downloader/downloader.go

```diff
@@ -951,6 +956,11 @@ func (d *Downloader) fetchNodeData() error {
 				syncStatsStateDone := d.syncStatsStateDone // Thread safe copy for the log below
 				d.syncStatsLock.Unlock()
 
+				// If real database progress was made, reset any fast-sync pivot failure
+				if progressed && atomic.LoadUint32(&d.fsPivotFails) > 1 {
```

Comment:
I assume the point of using atomic here is to guard against race conditions of some sort, but this if/then statement itself introduces a race condition, since the read/modify/write is not atomic.